### PR TITLE
feat: add highlight words feature for messages and user preferences 

### LINKED ISSUE #6371: feature: Highlight Keywords Missing in Mobile App
## Feature Description

In the Rocket.Chat web application, under user preferences, users can set up highlight keywords that notify them when specific words are mentioned. This feature is not currently available in the Rocket.Chat mobile app, resulting in a feature gap between platforms.

## Motivation and Use Case

Highlight keywords are essential for power users who rely on custom triggers to stay updated in busy channels. Without this feature on mobile, users may miss important messages when away from desktop.

## Implementation Ideas

Introduce a “Highlight Keywords” section under Preferences in the mobile app to mirror the functionality of the web version.

## Is this feature available in the API or web version?

✅ Available in Web Version

## Rocket.Chat Server Version

_No response_

## Device Name

Any

## OS Version

Any

## Additional Context

The web version has the "Highlights" section:  
Web Highlights

Mobile version :  
Mobile Missing Highlights

### PR DESCRIPTION
**Pull request by DSingh0304**
## Proposed changes
This PR implements the highlight words feature in the Rocket.Chat React Native app, bringing it in sync with the web version. Users can now set highlight words in their preferences, and matching words in messages will be highlighted with a red background.

**Key Changes:**
- Added highlight words input field in User Preferences view with save functionality
- Implemented regex-based text highlighting in message rendering
- Updated API calls to use correct field name (`highlights`) matching web version
- Added proper type definitions and translations for the feature

## How to test or reproduce
1. Navigate to User Preferences > Accessibility & Appearance
2. Add highlight words in the new "Highlight words" field (comma-separated)
3. Save preferences
4. Send or view messages containing the highlight words
5. Verify that matching words are highlighted with red background

##Screenshots

Screenshot from 2025-11-18 20-06-49
Screenshot from 2025-11-18 20-07-07
Screenshot from 2025-11-18 20-07-30

## Screen Recording

[Screencast from 2025-11-17 22-34-57.webm](https://github.com/user-attachments/assets/af60e3f2-bb36-4b13-9bd5-b5a0ae7033b2)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
**Files Modified:**
- `app/views/UserPreferencesView/index.tsx` - Added highlights UI and save logic
- `app/containers/markdown/components/Plain.tsx` - Implemented text highlighting
- `app/containers/markdown/contexts/MarkdownContext.ts` - Added highlights context
- `app/containers/markdown/index.tsx` - Passed highlights to context
- `app/containers/message/Content.tsx` - Propagated highlights to Markdown
- `app/containers/message/index.tsx` - Added highlight detection logic
- `app/containers/message/interfaces.ts` - Added highlights prop
- `app/definitions/IUser.ts` - Extended notification preferences
- `app/i18n/locales/en.json` - Added translation keys
- `app/views/RoomView/index.tsx` - Passed highlights from user settings

**Testing Results:**
✅ All existing tests pass  
✅ Markdown rendering tests: 11/11 passed  
✅ No breaking changes to existing functionality

Closes: #6371 

This feature aligns the mobile app with the Rocket.Chat web version's highlight words functionality.

## Summary by CodeRabbit

* **New Features**
  * Message highlighting: user-configurable words emphasized in messages (case-insensitive, exact-word, punctuation-aware, theme-aware colors); accessibility labels updated.

* **UI**
  * Preferences: new Highlights input with placeholder, description, dirty/save behavior, save button, optimistic save and success/failure toasts.
  * Highlights propagate to message lists and message rendering.

* **Localization**
  * Added strings for highlight labels, descriptions, and save feedback.

* **Tests**
  * Added tests covering highlight rendering and edge cases.

* **Documentation**
  * Story demonstrating highlight behavior added.

### CODE REVIEW

File: app/views/UserPreferencesView/index.tsx [hunk 1]
```diff
@@ -58,10 +73,77 @@ const UserPreferencesView = ({ navigation }: IUserPreferencesViewProps): JSX.Ele
...
+					}
+				} catch (err) {
+					log(err);
+					showToast(I18n.t('Highlights_save_failed'));
+				}
+			} catch (e) {
+				log(e);
+				showToast(I18n.t('Highlights_save_failed'));
+			}
+		};
```
[the remark this answers is not in the document]
Comment (DSingh0304) [reply app/views/UserPreferencesView/index.tsx:143]: Resolved in [7910c34](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/6809/commits/7910c34611413e7a95329e918ee49c31f778f50b)

File: app/views/UserPreferencesView/index.tsx [hunk 2]
```diff
@@ -116,6 +201,37 @@ const UserPreferencesView = ({ navigation }: IUserPreferencesViewProps): JSX.Ele
...
+					<FormTextInput
+						value={highlights}
+						onChangeText={value => {
+						setHighlights(value);
+						setDirty(value !== initialHighlightRef.current);
+					}}
+						onBlur={() => {
+						// Only save if dirty or changed
+						saveHighlights(highlights);
+					}}
```
[the remark this answers is not in the document]
Comment (DSingh0304) [reply app/views/UserPreferencesView/index.tsx:218]: Resolved in [f753a37](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/6809/commits/f753a37362adfa9dba4b361ad19c24bb8de9e01f)

**Review by OtavioStasiak — CHANGES_REQUESTED** [review]
Hey! Thank you for your contribution. Here are a few changes:

- Lint is failing.
- E2E tests are missing.
- Storybook is missing.

Please add screenshots to the PR description and include Storybook cases for message-validation scenarios, such as highlighted messages with `bold`, `italic` and `code`. We need to make sure all cases are working correctly.

**Comment by DSingh0304** [comment]
Yeah, sure I will make the changes and update the PR.

File: app/containers/markdown/components/Plain.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      </ThemeContext.Provider>
+    );
+
+    const h1 = tree.getByText('rocket');
+    const h2 = tree.getByText('world');
+    expect(h1).toBeTruthy();
+    expect(h2).toBeTruthy();
+    expect(getBackground(h1)).toBe(colors.statusBackgroundDanger);
+    expect(getBackground(h2)).toBe(colors.statusBackgroundDanger);
+  });
```
[the remark this answers is not in the document]
Comment (DSingh0304) [reply app/containers/markdown/components/Plain.test.tsx:74]: Fixed in [1589f1c](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/6809/commits/1589f1c37f001308a21707c29805212d6579996e)

**Comment by DSingh0304** [comment]
Hey, I have updated the PR and uploaded the screenshots. Please review once.
